core/src/nucleus/ribosome/api/call.rs

```diff
@@ -112,38 +114,22 @@ pub(crate) fn reduce_call(
         return;
     }
     let dna = state.dna.clone().unwrap();
-    let maybe_cap = get_capability_with_zome_call(&dna, &fn_call);
-    if let Err(fn_res) = maybe_cap {
+    let maybe_public = is_fn_public(&dna, &fn_call);
```

Comment:
Hm, I'm not sure if this check is actually correct here.
`Action::Call` is only used for calls that originate from a local `hdk::call`. So this is how we call zome functions of other zomes.

`Action::ExecuteZomeFunction` is what facilitates calls coming in from the outside..

Reply:
ok refactored the code that validates the call in both those actions..